### PR TITLE
ZER-144-create-feature-flag

### PR DESCRIPTION
**Pull request by Beerka1**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @opazyniuk

### Second Level Review

- [ ] @PavloMS

## Summary of issue

Add table feature flags and create interface to add and check status of existing feature flag.

## Summary of change

There were created feature flag table and 3 methods activate, deactivate and  activate? in FeatueFlag model.

## Testing approach

I used RSpec tests to cover my model.

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/helpers/feature_flags_helper.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+module FeatureFlagsHelper
+end
```
Comment (PavloMS) [inline app/helpers/feature_flags_helper.rb:4]: what is it used for?

File: app/models/feature_flag.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class FeatureFlag < ApplicationRecord
+  validates :name, presence: true, uniqueness: true
+
+  def add_new_feature
```
Comment (PavloMS) [inline app/models/feature_flag.rb:6]: def create

File: app/models/feature_flag.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    FeatureFlag.create(name: name, enabled: false)
+  end
+
+  def feature_added?
```
Comment (PavloMS) [inline app/models/feature_flag.rb:10]: def feature_exist?

File: spec/helpers/feature_flags_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+# describe FeatureFlagsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe FeatureFlagsHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (PavloMS) [inline spec/helpers/feature_flags_helper_spec.rb:17]: why do we need this?

File: spec/models/feature_flag_spec.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'pry'
```
Comment (PavloMS) [inline spec/models/feature_flag_spec.rb:4]: remove

File: spec/requests/feature_flags_spec.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe "FeatureFlags", type: :request do
+  describe "GET /index" do
+    pending "add some examples (or delete) #{__FILE__}"
+  end
+end
```
Comment (PavloMS) [inline spec/requests/feature_flags_spec.rb:7]: same as above

File: spec/helpers/feature_flags_helper_spec.rb (hunk 4 above)
Comment (Beerka1) [reply spec/helpers/feature_flags_helper_spec.rb:17]: File was created automatically, is there need to remove this file?

Comment (PavloMS) [reply spec/helpers/feature_flags_helper_spec.rb:17]: If you don't use it, than remove

**Review by AndriiStegnytskyi — APPROVED** [review]

**Review by PrincTwilig — APPROVED** [review]

File: spec/factories/feature_flags.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :feature_flag do
+    name {'FeatureFlag'}
+    enabled { false }
+  end
+end
```
Comment (opazyniuk) [inline spec/factories/feature_flags.rb:6]: newline

File: spec/models/feature_flag_spec.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      it { expect(subject.create.deactivate.enabled).to be_falsey }
+    end
+  end
+
+  describe '#active?' do
+    context 'when feature is not added in table' do
+      it 'return false' do
+        expect(subject.active?).to be_falsey
+      end
+    end
```
Comment (opazyniuk) [inline spec/models/feature_flag_spec.rb:42]: spaces between tests

File: app/models/feature_flag.rb [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class FeatureFlag < ApplicationRecord
+  validates :name, presence: true, uniqueness: true
+
+  def create
+    FeatureFlag.create(name: name, enabled: false)
```
Comment (opazyniuk) [inline app/models/feature_flag.rb:7]:
```suggestion
    FeatureFlag.create!(name: name, enabled: false)
```

**Review by opazyniuk — CHANGES_REQUESTED** [review]
left a few minor sugestions

**Review by opazyniuk — APPROVED** [review]

**Pull request merged**